From ac3944b364fd14ecc450ae178cdfc91697c6050a Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 7 Apr 2025 22:29:50 +0800
Subject: [PATCH 04/21] AOSCOS: media: enable VA-API decoding/encoding by 
 default

Enable the following features by default:

- AcceleratedVideoEncoder
- AcceleratedVideoDecoder
- AcceleratedVideoDecodeLinuxGL
- AcceleratedVideoDecodeLinuxZeroCopyGL
- VaapiOnNvidiaGPUs

Link: https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/gpu/vaapi.md
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 media/base/media_switches.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/media/base/media_switches.cc b/media/base/media_switches.cc
index 1cdacae..3faba70 100644
--- a/media/base/media_switches.cc
+++ b/media/base/media_switches.cc
@@ -751,7 +751,7 @@ BASE_FEATURE(kAcceleratedVideoDecodeLinuxGL, base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kAcceleratedVideoEncodeLinux,
              "AcceleratedVideoEncoder",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Ignore the non-intel driver blacklist for VaapiVideoDecoder implementations.
 // Intended for manual usage only in order to gague the status of newer driver
@@ -761,7 +761,7 @@ BASE_FEATURE(kVaapiIgnoreDriverChecks, base::FEATURE_DISABLED_BY_DEFAULT);
 
 // NVIDIA VA-API drivers do not support Chromium and can sometimes cause
 // crashes, disable VA-API on NVIDIA GPUs by default. See crbug.com/1492880.
-BASE_FEATURE(kVaapiOnNvidiaGPUs, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kVaapiOnNvidiaGPUs, base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Enable VA-API hardware low power encoder for all codecs on intel Gen9x gpu.
 BASE_FEATURE(kVaapiLowPowerEncoderGen9x, base::FEATURE_ENABLED_BY_DEFAULT);
-- 
2.54.0

